lib/rouge/lexers/ttcn3.rb

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class TTCN3 < RegexLexer
+      title "TTCN3"
+      desc "The TTCN3 programming language (ttcn-3.org)"
+
+      tag 'ttcn3'
+      filenames '*.ttcn', '*.ttcn3'
+      mimetypes 'text/x-ttcn3', 'text/x-ttcn'
+
+      def self.keywords
+        @keywords = %w(
+          module import group type port component signature external
+          execute const template function altstep testcase var timer if
+          else select case for while do label goto start stop return
+          break int2char int2unichar int2bit int2enum int2hex int2oct
+          int2str int2float float2int char2int char2oct unichar2int
+          unichar2oct bit2int bit2hex bit2oct bit2str hex2int hex2bit
+          hex2oct hex2str oct2int oct2bit oct2hex oct2str oct2char
+          oct2unichar str2int str2hex str2oct str2float enum2int
+          any2unistr lengthof sizeof ispresent ischosen isvalue isbound
+          istemplatekind regexp substr replace encvalue decvalue
+          encvalue_unichar decvalue_unichar encvalue_o decvalue_o
+          get_stringencoding remove_bom rnd hostid send receive
+          setverdict
+        )
+      end
+
+      def self.reserved
+        @reserved = %w(
+          all alt apply assert at configuration conjunct const control
+          delta deterministic disjunct duration fail finished fuzzy from
+          history implies inconc inv lazy mod mode notinv now omit
+          onentry onexit par pass prev realtime seq setstate static
+          stepsize stream timestamp until values wait
+        )
+      end
+
+      def self.types
+        @types = %w(
+          anytype address boolean bitstring charstring hexstring octetstring
+          component enumerated float integer port record set of union universal
+        )
+      end
+
+      # optional comment or whitespace
+      ws = %r((?:\s|//.*?\n|/[*].*?[*]/)+)
+      id = /[a-zA-Z_][a-zA-Z0-9_]*/
+      const_name = /[A-Z][a-zA-Z0-9_]*\b/
+      module_name = /[A-Z][a-zA-Z0-9_]*\b/
+
+      state :root do
+        rule %r/\s+/, Text
```

Comment:
The problem with the previous version was that I'd removed your newline rule at the end of the `:root` state but I'd forgotten to change this rule. You originally had it as `rule %r/[^\S\n]/, Text`. Is there a reason you were structuring it like this?

Reply:
@YannGarcia wrote:

> Not really, I just reused my existing local Rouge code. The point is that
> until now I was using it internally but it was requested me to add it in
> ETSI forge which is based on GitLab.
> This is why I sent you the Pull request.

Sorry, my question was a little unclear. What I meant was that `/[\S\n]+/` will match all whitespace characters _except_ `\n`. That seems like an unusual choice and I wondered if it was intentional?

Reply:
@YannGarcia wrote:

> No, I think it's my mistake :(

That's OK :) Fixed now!